design/k8s_secrets_labels_design.md

```diff
@@ -0,0 +1,218 @@
+# K8s Secrets Labels Design
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+## Glossary
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Label](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) | Key/value pairs that are attached to K8s objects. They are used to identify and group objects for effective querying and selection. |
+
+## Background
+
+As part of Phase 2 for Secrets Provider for K8s effort, we want to improve the customer experience by offering the ability to use labels for easy identification of K8s secrets whose secret values should come from Conjur. This will allow customers to list in terms of groupings instead of having to list the K8s secrets individually.
+
+## Issue description
+
+At current, the Secrets Provider code looks at the `K8S_SECRETS` environment variable in each of the pod manifests to know which K8s Secret to update with Conjur values. 
```

Comment:
what do you mean by `in each of the pod manifests`? the secrets-provider has one manifest that defines its container and it looks at that manifest.

Reply:
i think that an example of a manifest snippet can help those who are not familiar with it.

Reply:
if a customer has multiple secrets providers, then there would be multiple manifests. I will add an example